File: app/src/main/java/com/abs/launcher/util/BitmapCache.java
package com.abs.launcher.util;

import android.graphics.Bitmap;

import com.abs.launcher.util.cache.SoftCache;

public class BitmapCache extends SoftCache<String, Bitmap> {

    public BitmapCache() {
        super();
    }

    public BitmapCache(int size) {
        super(size);
    }

    @Override
    protected void onClear(Bitmap object) {
        if (object != null && !object.isRecycled()) {
            object.recycle();
        }
    }
}

File: app/src/main/java/com/abs/launcher/util/BitmapCache.kt
package com.abs.launcher.util

import android.graphics.Bitmap
import com.abs.launcher.util.cache.SoftCache

/**
 * Created by zy on 17-12-17.
 */
class BitmapCache : SoftCache<String, Bitmap> {
    constructor() : super()
    constructor(initialSize: Int) : super(initialSize)

    override fun onClear(value: Bitmap) {
        if (!value.isRecycled) {
            value.recycle()
        }
    }
}